docs/reference/variables.rst

```diff
@@ -0,0 +1,49 @@
+.. _variables:
+
+=========
+Variables
+=========
+
+.. currentmodule:: dimod.variables
+
+.. automodule:: dimod.variables
+
+Variables Class
+===============
+
+.. autoclass:: Variables
+
+Properties
+----------
+
+.. autosummary::
+   :toctree: generated/
+
+    ~Variables.is_range
+
+Methods
+-------
+
+.. autosummary::
+   :toctree: generated/
+
+    ~Variables.index
+    ~Variables.count
+    ~Variables.to_serializable
+
+Mutation Methods
```

Comment:
Can the internal nature be more prominent? Maybe add Private to the title or at least bold the warning below.

Reply:
All the methods are prepended by a `_` which I think is a pretty clear signal in a code context. In the docs context, I am not sure bolding it would make it any more likely to be read. It's not like this section is hidden among a lot of other text. We could say "semi-private" or something, but they are not strictly speaking private...